api/opencl_runtime_layer.asciidoc

```diff
@@ -13915,16 +13915,33 @@ of 0 or 1.
 The simultaneous use capability removes this restriction and allows
 command-buffers to have a <<pending_count, Pending Count>> greater than 1.
 
-[[compatible]]
 Command-buffers are created using an ordered list of command-queues that
-commands are recorded to and execute on by default.
-These command-queues can be replaced on command-buffer enqueue with
-different command-queues, provided for each element in the replacement list
-the substitute command-queue is compatible with the command-queue used on
-command-buffer creation.
-A _compatible_ command-queue is defined as a command-queue with
-identical properties targeting the same device and in the same OpenCL
-context.
+commands are recorded to and execute on by default. All these queue objects
+must share the same context, but may be associated with different devices when
+the {cl_khr_command_buffer_multi_device_EXT} extension is supported.
+
+When constructing a command-buffer by appending commands, the queue parameter
+passed for the command being created is used to set the device with which the
+command will be associated with, and also inform the scheduling of the command.
+If the queue is an in-order queue, then an additional dependency is created on the
+last command appended to the command-buffer using the same queue parameter. If
+the queue is an out-of-order queue, then no extra dependencies on previous
+commands using the same queue are created. All queue properties other than
+{CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE} are ignored for the purposes of command
+creation, with the exception of any vendor extension defined queue properties
+that explcitly define semantics for this purpose.
+
+The command-queues used on command-buffer creation must be replaced on
```

Comment:
Why say "must be replaced"? Can't the user use the same queue objects for creation and execution?

Reply:
I'll try reword this. I was trying to communicate that since the queues used on command-buffer creation are really placeholders used to inform the device and dependencies of commands, those placeholders must be replaced with queues which will actually submit work to the device.

Reply:
I understand, but the question is: **_must_** the queues used for recording and replaying be different? If yes, then your wording is OK; if not, you should replace "must be replaced" with "may be replaced".

Reply:
There is no requirement that the queues **must** be different, they only **may** be different as I say in the next sentence, so I'll reword this sentence.

Reply:
Changed this line to say "When enqueuing a command-buffer, a list of command-queues to execute the command-buffer on can be passed by the user, otherwise the command-queues set on command-buffer creation are used by default for execution. A user passed list may contain different command-queues, ..."